packages/ssz/src/core/createValue.ts

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import assert from "assert";
+import {AnySSZType, FullSSZType, parseType, Type} from "@chainsafe/ssz-type-schema";
+
+import {assertValidValue} from "./assertValidValue";
+import {defaultValue} from "./defaultValue";
+
+/**
+ * Given an ssz type, create an empty value, optionally filled in with a default value
+ *
+ * The default value can be only partially specified.
+ */
+export function createValue<T>(type: AnySSZType<T>, value: any = null): T {
+  return _createValue(parseType(type), value) as T;
+}
+
+function _createValue(type: FullSSZType, value: any = null): any {
```

Comment:
why aren't you using generics here as well?

Reply:
not really needed at this lower level, only the higher level functions are used regularly.
I remember attempting this at some point and running into trouble, and thinking it's not worth it.